### ISSUE TITLE
Additional sql query with junction table

### ISSUE THREAD

**Issue by krebse** [yiisoft/yii2#12390]
### What steps will reproduce the problem?

My project contain three table : post , category , category_post_assignment

I get ID of the category and search for related posts

```
$category = Category::find()
            ->with('posts')
            ->where(['id' => 2])
            ->one();
```

```
$category = Category::find()
            ->where(['id' => 2])
            ->one();

$posts = $category->posts;
```

In the normal first query get category from relevant table
`SELECT * FROM `tbl_category` WHERE `id`=2`

second query get relevant posts id from assignment table
`SELECT * FROM `tbl_category_post_assignment` WHERE `category_id`=2`

third query get the posts from post table
`SELECT * FROM `tbl_post` WHERE `id` IN (1, 2, ....)`

The issue is after second query if don't find relevant post the third query still execute in additional query.
`SELECT * FROM `tbl_post` WHERE 0=1`
### What is the expected result?

When don't find relevant post in second query the third query shouldn't be execute.
### Additional info

| Q | A |
| --- | --- |
| Yii version | 2.0.9 |
| PHP version | 7.0 |

**Comment by klimov-paul** [comment]
Possible fix: #12708

**Comment by cebe** [comment]
Fixed, thanks for reporting!
